File: cloud-manager-web/src/main/resources/static/js/service/specificationService.js
//服务层
app.service('specificationService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../Specification/specification/findAll');
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../Specification/specification/findPage?findPage='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../Specification/specification/findOne?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.put('../Specification/specification/',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../Specification/specification/',entity );
	}
	//删除
	this.dele=function(ids){
		console.log(ids)
		return $http.get('../Specification/specification/delete?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../Specification/specification/search?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: cloud-sellergoods-service/src/main/java/com/jessie/mall/sellergoodsService/service/SpecificationService.java
package com.jessie.mall.sellergoodsService.service;

import com.jessie.mall.common.common.TypeTemplateResult;
import com.jessie.mall.common.common.pageResult;
import com.jessie.mall.pojo.TbSpecification;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface SpecificationService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbSpecification> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public pageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbSpecification specification);
	
	
	/**
	 * 修改
	 */
	public void update(TbSpecification specification);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbSpecification findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public pageResult findPage(TbSpecification specification, int pageNum, int pageSize);

    List<TypeTemplateResult> findforType();
}
